x/delayedack/keeper/rollapp_packet.go

```diff
@@ -65,12 +65,15 @@ func (k Keeper) UpdateRollappPacketTransferAddress(
 	}
 	// Set the recipient and sender based on the rollapp packet type
 	recipient, sender := transferPacketData.Receiver, transferPacketData.Sender
+	var originalRecipient string
 	switch rollappPacket.Type {
 	case commontypes.RollappPacket_ON_RECV:
+		originalRecipient = recipient
 		recipient = address
 	case commontypes.RollappPacket_ON_TIMEOUT:
 		fallthrough
 	case commontypes.RollappPacket_ON_ACK:
+		// TODO: currently we only save the original recipient for RECV, but we could also do it for TIMEOUT and ACK for symmetry if needed
```

Comment:
I would use this `rollappPacket.OriginalRecvReceiver` to keep the original address in all cases (recv/ack/timeout).

it would be used only on `OnRecv`, but it's benefical for queries